src/storage/mongodb/UserStorage.ts

```diff
@@ -658,13 +658,20 @@ export default class UserStorage {
       });
     }
     // Add Tags
-    if (params.tagIDs) {
+    if (!Utils.isEmptyArray(params.tagIDs) || !Utils.isEmptyArray(params.visualTagIDs)) {
       DatabaseUtils.pushTagLookupInAggregation({
         tenantID, aggregation, localField: '_id', foreignField: 'userID', asField: 'tag'
       });
-      aggregation.push({
-        $match: { 'tag.id': { $in: params.tagIDs } }
-      });
+      if (!Utils.isEmptyArray(params.tagIDs)) {
```

Comment:
Do we still need to keep the tagIDs param?

Reply:
No, done.